plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/CloudSdkUpdateNotification.java

```diff
@@ -0,0 +1,118 @@
+import org.eclipse.mylyn.commons.ui.dialogs.AbstractNotificationPopup;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.ui.IWorkbench;
+
+/** A notification that a new version of the Cloud SDK is available. */
+public class CloudSdkUpdateNotification extends AbstractNotificationPopup {
+  private static final Logger logger = Logger.getLogger(CloudSdkUpdateNotification.class.getName());
+
+  private static final String CLOUD_SDK_RELEASE_NOTES_URL =
+      "https://cloud.google.com/sdk/docs/release-notes";
+
+  /** Show a notification that an update is available. */
+  public static void showNotification(
+      IWorkbench workbench, CloudSdkVersion currentVersion, Runnable updateTrigger) {
+    workbench
+        .getDisplay()
+        .asyncExec(
+            () -> {
+              CloudSdkUpdateNotification popup =
+                  new CloudSdkUpdateNotification(workbench, currentVersion, updateTrigger);
+              popup.open();
+            });
+  }
+
+  private IWorkbench workbench;
+  private CloudSdkVersion sdkVersion;
+  private Runnable updateRunnable;
+  private Image gcpImage;
+
+  private CloudSdkUpdateNotification(
+      IWorkbench wb, CloudSdkVersion currentVersion, Runnable triggerUpdate) {
+    super(wb.getDisplay());
+    workbench = wb;
+    sdkVersion = currentVersion;
+    updateRunnable = triggerUpdate;
+  }
+
+  @Override
+  protected String getPopupShellTitle() {
+    return Messages.getString("CloudSdkUpdateNotificationTitle");
+  }
+
+  @Override
+  protected Image getPopupShellImage(int maximumHeight) {
+    if (gcpImage == null) {
+      gcpImage = SharedImages.CLOUDSDK_IMAGE_DESCRIPTOR.createImage(workbench.getDisplay());
+    }
+    return gcpImage;
+  }
+
+  @Override
+  protected void createContentArea(Composite parent) {
+    Link message = new Link(parent, SWT.WRAP);
+    message.setText(Messages.getString("CloudSdkUpdateNotificationMessage", sdkVersion));
+    message.addSelectionListener(
+        new SelectionAdapter() {
+          @Override
+          public void widgetSelected(SelectionEvent event) {
+            switch (event.text) {
+              case "update":
+                updateRunnable.run();
+                break;
+              case "releasenotes":
+                showReleaseNotes();
+                break;
+              default:
+                logger.warning("Unknown selection event: " + event.text);
+                break;
+            }
+          }
+        });
+  }
+
+  protected void showReleaseNotes() {
+    WorkbenchUtil.openInBrowser(
+        workbench,
+        CLOUD_SDK_RELEASE_NOTES_URL,
+        "com.google.cloud.tools.sdk.releasenotes",
+        null,
+        null);
+  }
+
+  @Override
+  protected void handleShellCloseEvent() {
```

Comment:
I remember I had some surprising behavior with this method. Let me try to recall that.

Reply:
In my testing, clicking on the "X" button at the top-right corner or pressing Escape doesn't call this. (The Escape behavior seems contradicting its Javadoc honestly).

There were also other cases where this is not called: #2622

Reply:
Hmm nice find.  Took a quick look to see how to fix this and discovered that the `AbstractNotificationPopup` uses a ResourceManager which is cleaned up by its `close()` method, and which is called on the X and other dismisses.  And simplifies this code.